### ISSUE TITLE
using context variable 'size' in templates

### ISSUE THREAD

**Issue by Darkvater** [thymeleaf/thymeleaf#94]
Using ThymeLeaf 2.0.13
I have created a template variable in ThymeLeaf called "size". When I want to show this value its actual value is not shown. After a bit of looking around I figured out it shows the context size. Is this correct? If so, where is this documented?
A quick example:

``` java
Context context = new Context();
context.setVariable("variable1", "hello world");
context.setVariable("size", BigDecimal.TEN);
```

In the template if I now do:

``` html
<html><body><span th:text="${size}">this should show TEN</span></body></html>
```

It shows 2, and not 10. Two is the size of the number of objects in the context.

**Comment by danielfernandez** [comment]
If a "size" variable exists, it should supersede the map's "size" property. Queued for 2.1

**Comment by danielfernandez** [comment]
Fixed and updated 2.0.16-SNAPSHOT

**Comment by rosko01** [comment]
The fix for this issue introduced a hard dependency on OGNL.  Before this change, users of the Spring-Dialect (which defaults to SpEL) could omit the ognl.jar and javassist.jar from their list of dependencies.

Please reconsider.
